src/metabase/query_processor/card.clj

```diff
@@ -220,7 +220,6 @@
                           :visualization-settings (:visualization_settings card)}
                    (and (= (:type card) :model) (seq (:result_metadata card)))
                    (assoc :metadata/model-metadata (:result_metadata card)))]
-    (api/check-not-archived card)
```

Comment:
Note that this hasn't ever actually checked anything, because we select the card with an itemized list of columns and `:archived` isn't one of them :sob:

Reply:
Should we check it, though? Ought we to add :archived to the columns we select from card?

Reply:
One of the goals here was to allow people to view cards that had been archived, so I think we need to remove this check. I did add a similar check to the /public API, to prevent running public cards that had been archived - but generally speaking it should be allowable to process an archived card.